source/on.js

```diff
@@ -0,0 +1,31 @@
+import curryN from './internal/_curryN';
+
+
+/**
+ * Takes a binary curried function `f`, a unary function `g`, and two values.
+ * Applies `g` to each value, then applies the result of each to `f`.
+ *
+ * Also known as the P combinator.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig (b -> b -> c) -> (a -> b) -> a -> a -> c
```

Comment:
I actually just stole the signature from Sanctuary, I haven't much experience with HM types. I'm not sure I even understand what could be ordered differently

Reply:
The point is not the signature block, although that would have to change as well.

This is just switching the first two parameters.  It's the difference between 

```js
curry (f, g, a, b) => f ( g(a ) ) ( g(b) ) )
```

and 

```js
curry (g, f, a, b) => f ( g(a ) ) ( g(b) ) )
```

which would change the signature from

```hs
on :: (b -> b -> c) -> (a -> b) -> a -> a -> c
```

to  

```hs
on :: (a -> b) -> (b -> b -> c) -> a -> a -> c
```

Reply:
> I actually just stole the signature from Sanctuary

Heh, I guess I know where I stood on this a little while ago at least 😛 .

I am OK with the merging function going first as it matches with current behavior of converge and useWith. The other way draws me due to `on(prop('date'), sort)` reading better to me than `on(sort, prop('date'))`, but I am OK with it as it stands.

Reply:
Swapping position of `g` and `f` params seems really odd to me? It would also break expectations if you know `on` from Haskell or Sanctuary..

Maybe under a different name in that case? `by` might also work.